pkg/discovery/network/generic.go

```diff
@@ -62,13 +59,122 @@ func discoverGenericNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, er
 }
 
 func findClusterIPRange(clientSet kubernetes.Interface) (string, error) {
+	clusterIPRange, err := findClusterIPRangeFromApiserver(clientSet)
+	if err != nil || clusterIPRange != "" {
+		return clusterIPRange, err
+	}
+
+	clusterIPRange, err = findClusterIPRangeFromServiceCreation(clientSet)
+	if err != nil || clusterIPRange != "" {
+		return clusterIPRange, err
+	}
+
+	return "", nil
+}
+
+func findClusterIPRangeFromApiserver(clientSet kubernetes.Interface) (string, error) {
 	return findPodCommandParameter(clientSet, "component=kube-apiserver", "--service-cluster-ip-range")
 }
 
+func findClusterIPRangeFromServiceCreation(clientSet kubernetes.Interface) (string, error) {
+	// find service cidr based on https://stackoverflow.com/questions/44190607/how-do-you-find-the-cluster-service-cidr-of-a-kubernetes-cluster
+	invalidSvcSpec := &v1.Service{
+		ObjectMeta: v1meta.ObjectMeta{
+			Name: "invalid-svc",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "1.1.1.1",
+			Ports: []v1.ServicePort{
+				{
+					Port: 443,
+					TargetPort: intstr.IntOrString{
+						IntVal: 443,
+					},
+				},
+			},
+		},
+	}
+
+	// decide which namespace to create the service
+	ns := os.Getenv("WATCH_NAMESPACE")
+	// use "submariner-operator" if WATCH_NAMESPACE is "submariner-operator" (in operator)
+	if ns != "submariner-operator" {
+		// otherwise, use "default" (in subctl command)
+		ns = "default"
+	}
+
+	// create service to the namespace
+	_, err := clientSet.CoreV1().Services(ns).Create(invalidSvcSpec)
+
+	// creating invalid service didn't fail as expected
+	if err == nil {
+		return "", fmt.Errorf("creating invalid service(%v) didn't fail", invalidSvcSpec)
```

Comment:
The callers of `discoverGenericNetwork` are as follows:

```
[Operator]
Reconcile
  discoverNetwork
    getClusterNetwork
      Discover
        discoverGenericNetwork

[Command line]
getNetworkDetails
  Discover
    discoverGenericNetwork
  if err != nil {
    status.QueueWarningMessage(fmt.Sprintf("Error trying to discover network details: %s", err))
  }
```

So, as for command line case, the error is already been passed to `status.QueueWarningMessage`.

And, other functions called from `discoverGenericNetwork`, like `findClusterIPRangeFromApiserver` actually returns error and not catching it with `status.QueueWarningMessage`.

Therefore, the code should be already as you expected.

Reply:
My point was about whether it should return an error in this case, ie should we treat it like we didn't find the data similar to the other discovery methods. So for `subclt join`, this would at least allow it to prompt the user as a last resort. After all, this is just another attempt to find the data. This seems reasonable but it would be beneficial to print a warning. @mangelajo WDYT?

Reply:
Waiting for @mangelajo 's comment. But I'm now inclined to return nil and log error here, only for this case.

Meanwhile, I would like to confirm that it is safe to call `status.QueueWarningMessage` outside CLI's context, or from operator. It looks like a package under `pkg/internal/cli` and I'm not familiar with its implementation details.

Reply:
Unfortunately `status.QueueWarningMessage` wouldn't be appropriate in the operator. The operator uses a different logging API. We could pass in a function to log warnings.

Reply:
Added logging code in `controllers/submariner/submariner_networkdiscovery.go` for operator to log it (See the first commit). Also, re-organized and squashed commits to review easier.